scubagoggles/rego/Drive.rego

```diff
@@ -0,0 +1,1588 @@
+    "Criticality": "Shall",
+    "ReportDetails": utils.ReportDetails(NonCompliantOUs1_8, NonCompliantGroups1_8),
+    "ActualValue": {"NonCompliantOUs": NonCompliantOUs1_8, "NonCompliantGroups": NonCompliantGroups1_8},
+    "RequirementMet": Status,
+    "NoSuchEvent": false
+}
+if {
+    Check1_8_OK
+    Conditions := {count(NonCompliantOUs1_8) == 0, count(NonCompliantGroups1_8) == 0}
+    Status := (false in Conditions) == false
+}
+#--
+
+###################
+# GWS.DRIVEDOCS.2 #
+###################
+
+#
+# Baseline GWS.DRIVEDOCS.2.1
+#--
+
+DriveId2_1 := utils.PolicyIdWithSuffix("GWS.DRIVEDOCS.2.1")
+
+LogMessage2_1 := "Shared Drive Creation new_team_drive_admin_only"
+
+Check2_1_OK if {
+    not PolicyApiInUse
+    events := utils.FilterEventsOU(LogEvents, LogMessage2_1, utils.TopLevelOU)
+    count(events) > 0
+}
+
+Check2_1_OK if {PolicyApiInUse}
+
+NonComplianceMessage2_1 := "Members with manager access can override shared drive settings."
+
+NonCompliantOUs2_1 contains {
+    "Name": OU,
+    "Value": NonComplianceMessage2_1
+}
+if {
+    not PolicyApiInUse
+    some OU in utils.OUsWithEvents
+    Events := utils.FilterEventsOU(LogEvents, LogMessage2_1, OU)
+    count(Events) > 0
+    LastEvent := utils.GetLastEvent(Events)
+    contains("true", LastEvent.NewValue) == false
+    LastEvent.NewValue != "DELETE_APPLICATION_SETTING"
+}
+
+NonCompliantGroups2_1 contains {
+    "Name": Group,
+    "Value": NonComplianceMessage2_1
+}
+if {
+    not PolicyApiInUse
+    some Group in utils.GroupsWithEvents
+    Events := utils.FilterEventsGroup(LogEvents, LogMessage2_1, Group)
+    count(Events) > 0
+    LastEvent := utils.GetLastEvent(Events)
+    contains("true", LastEvent.NewValue) == false
+    LastEvent.NewValue != "DELETE_APPLICATION_SETTING"
+}
+
+NonCompliantOUs2_1 contains {
+    "Name": OU,
+    "Value": NonComplianceMessage2_1
+}
+if {
+    some OU, settings in input.policies
+    DriveEnabled(OU)
+    managerOverride := settings.drive_and_docs_shared_drive_creation.allowManagersToOverrideSettings
+    managerOverride != false
+}
+
+tests contains {
+    "PolicyId": DriveId2_1,
+    "Criticality": "Should",
+    "ReportDetails": utils.NoSuchEventDetails(DefaultSafe, utils.TopLevelOU),
+    "ActualValue": "No relevant event for the top-level OU in the current logs",
+    "RequirementMet": DefaultSafe,
+    "NoSuchEvent": true
+}
+if {
+    not PolicyApiInUse
+    DefaultSafe := false
+    not Check2_1_OK
+}
+
+tests contains {
+    "PolicyId": DriveId2_1,
+    "Criticality": "Should",
+    "ReportDetails": utils.ReportDetails(NonCompliantOUs2_1, NonCompliantGroups2_1),
+    "ActualValue": {"NonCompliantOUs": NonCompliantOUs2_1, "NonCompliantGroups": NonCompliantGroups2_1},
+    "RequirementMet": Status,
+    "NoSuchEvent": false
+}
+if {
+    Check2_1_OK
+    Conditions := {count(NonCompliantOUs2_1) == 0, count(NonCompliantGroups2_1) == 0 }
+    Status := (false in Conditions) == false
+}
+#--
+
+#
+# Baseline GWS.DRIVEDOCS.2.2
+#--
```

Comment:
The report fails you no matter what you select for "Allow users outside... to access files in shared drives"

Reply:
![image](https://github.com/user-attachments/assets/0b693f7c-a47a-4860-96a0-3be7f68f1028)

Reply:
After implementing Google's policy "reduction" process, this is working for me
when I change settings in the Admin UI.